src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -386,7 +388,19 @@ public void write() {
             }
           case SHOULD_ROTATE:
             updateRotationTimers(currentRecord);
-            closeTempFile();
+            try {
```

Comment:
I think this is the wrong place to handle this. There's already a `try/catch` outside, which implements retry. This code is completely discarding retries, and handles _only_ this case. Remember that those files are on the network, so failures can be intermittent and recoverable.

Instead, what should be done is limit the _amount_ of retries in the outer catch, falling back recovery and/or abending.

Reply:
OK it would be very helpful to get more inputs for this.
Have below concerns or approaches. 
ConnectException is being thrown even at places like appending wal and not just at closeTempFile.
So as of now, upon any ConnectException only the retryTimeout is updated and break is down within the outer catch block.

If this handling is moved to outer catch block, is it okay to do a recovery and returning from the write() for ConnectException occurring at all of those instances ?
If not, should we have an extra exception handling within closeTempFile() to check if there is a failure during temp file closure and use a boolean flag to indicate need for recovery only in that instance ?

Your inputs would be really helpful. Thanks

Reply:
In my opinion the recover is needed for _all_ cases, since, presently, _any_ non-intermittent failure will put the TPW into an infinite loop.